cmd/controlplane/controller.go

```diff
@@ -273,11 +275,20 @@ func (o *controllerOptions) setupReconcilers(
 	promotionsReconcilerCfg promotions.ReconcilerConfig,
 	stagesReconcilerCfg stages.ReconcilerConfig,
 ) error {
+	var argoCDClient client.Client
+	if argocdMgr != nil {
+		argoCDClient = argocdMgr.GetClient()
+	}
+
+	directivesEngine := directives.NewSimpleEngine(credentialsDB, kargoMgr.GetClient(), argoCDClient)
+	promoMechanisms := promotion.NewMechanisms(kargoMgr.GetClient(), argoCDClient, credentialsDB)
```

Comment:
I actually love seeing the two of these side-by-side here and knowing half of it goes away soon. 😄